### PR TITLE
Move `aws-http` types into `aws-runtime`

### PR DESCRIPTION
**Pull request by jdisanti**
This issue addresses a semver compatibility problem similar to the one described in https://github.com/smithy-lang/smithy-rs/pull/3318, except for the storable types in the aws-http crate. I opted to move all of aws-http into aws-runtime since there was so little in there.

----

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

**Comment by rcoh** [comment]
how does this not break current clients that depend on aws-http? I guess we would MV or delete aws-http?

**Comment by jdisanti** [comment]
> how does this not break current clients that depend on aws-http? I guess we would MV or delete aws-http?

I think this will need to get released as a new minor version of aws-http so that it's not picked up automatically.

**Comment by rcoh** [comment]
we should carefully audit everything that's `pub` in here and consider if we want it put in `aws-runtime`. I suspect much of it doesn't need to be anymore since we removed the crate boundary?

File: aws/rust-runtime/aws-http/external-types.toml [hunk 1]
```diff
@@ -1,11 +1,3 @@
 allowed_external_types = [
-    "aws_smithy_runtime_api::http::headers::Headers",
```
Comment (rcoh) [inline aws/rust-runtime/aws-http/external-types.toml:2]: is this actually true now? I thought the crate was empty

File: aws/rust-runtime/aws-http/Cargo.toml [hunk 2]
```diff
@@ -2,24 +2,13 @@
 name = "aws-http"
 version = "0.0.0-smithy-rs-head"
```
Comment (rcoh) [inline aws/rust-runtime/aws-http/Cargo.toml:3]: I think you need to change this version to some sort of EOL version? we need to be careful **not** to publish this with the next set of unstable crates

File: aws/rust-runtime/aws-http/external-types.toml (hunk 1 above)
Comment (jdisanti) [reply aws/rust-runtime/aws-http/external-types.toml:2]: The crate re-exports the types that were moved into aws-runtime.

File: aws/rust-runtime/aws-http/Cargo.toml (hunk 2 above)
Comment (jdisanti) [reply aws/rust-runtime/aws-http/Cargo.toml:3]: Good call. Updated.

**Comment by jdisanti** [comment]
> we should carefully audit everything that's pub in here and consider if we want it put in aws-runtime. I suspect much of it doesn't need to be anymore since we removed the crate boundary?

I took a pass over the types, and I don't see any opportunities here. The most we could do is choose not to remove/re-export something that's currently in aws-http, but everything looks to be used across crate boundaries.

**Review by rcoh — APPROVED** [review]

**Review by rcoh — APPROVED** [review]

**Pull request merged**